src/librustc_mir/interpret/validity.rs

```diff
@@ -608,47 +608,50 @@ impl<'rt, 'mir, 'tcx, M: Machine<'mir, 'tcx>> ValueVisitor<'mir, 'tcx, M>
                     return Ok(());
```

Comment:
Might be a better fix to just change this here to `len == 0 || layout.is_zst()`?

Reply:
Done. I made it two separate `if` statements because otherwise we'd query the element layout even for empty arrays and slices